operator/pkg/controllers/etcd/statefulset.go

```diff
@@ -0,0 +1,64 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/awslabs/kit/operator/pkg/apis/infrastructure/v1alpha1"
+	"github.com/awslabs/kit/operator/pkg/utils/object"
+	"github.com/awslabs/kit/operator/pkg/utils/patch"
+
+	"github.com/aws/aws-sdk-go/aws"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (c *Controller) reconcileStatefulSet(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
+	// Generate the default pod spec for the given control plane, if user has
+	// provided custom config for the etcd pod spec, patch this user
+	// provided config to the default spec
+	etcdSpec, err := patch.PodSpec(podSpecFor(controlPlane), controlPlane.Spec.Etcd.Spec)
```

Comment:
Given that we control the types on both pod specs, can we ensure that you can't runtime errors of the sort on line 37? Be squashing this error case, this simplifies a bunch of things. For one, it allows you do inline everything on line 39, including the full pod spec, which could be decorated inline. 
```
spec: patch.PodSpec(&PodSpec{ ... }, controlplane.Spec.EtcdSpec)
```
I guess it's not perfectly clean, since it's a bit awkward to have the `controlplane.Spec.EtcdSpec` dangling, but it doesn't make sense to put it as the first parameter w/ the patch semantic.

Reply:
Keeping this as is for now, I tried changing it but not sure, if we can avoid an error while patching.